packages/eas-cli/src/vcs/clients/git.ts

```diff
@@ -49,16 +51,30 @@ export default class GitClient extends Client {
   public async commitAsync({
     commitMessage,
     commitAllFiles,
+    nonInteractive = false,
   }: {
     commitMessage: string;
     commitAllFiles?: boolean;
+    nonInteractive?: boolean;
   }): Promise<void> {
+    await ensureGitConfiguredAsync({ nonInteractive });
+
     if (commitAllFiles) {
       await spawnAsync('git', ['add', '-A']);
     }
 
     await spawnAsync('git', ['add', '-u']);
-    await spawnAsync('git', ['commit', '-m', commitMessage]);
+    try {
```

Comment:
consider wrapping this whole function in try catch, I'm not aware of any case where git add could fail but if we already have logic for handling this error, then moving try to start of this function does not cost us anything and might catch future problems